psi4/driver/qcdb/molecule.py

```diff
@@ -1336,6 +1336,96 @@ def run_dftd3(self, func: str = None, dashlvl: str = None, dashparam: Dict = Non
         elif derint == 1:
             return jobrec['extras']['qcvars']['DISPERSION CORRECTION GRADIENT']
 
+    def run_dftd4(self, func=None, dashlvl=None, dashparam=None, dertype=None, verbose=1):
+        """Compute dispersion correction via Grimme's DFTD4 program.
+
+        Parameters
+        ----------
+        func : str, optional
+            Name of functional (func only, func & disp, or disp only) for
+            which to compute dispersion (e.g., blyp, BLYP-D2, blyp-d3bj,
+            blyp-d3(bj), hf+d). Any or all parameters initialized
+            from `dashcoeff[dashlvl][func]` can be overwritten via
+            `dashparam`.
+        dashlvl : str, optional
+            Name of dispersion correction to be applied (e.g., d, D2,
+            d3(bj), das2010). Must be key in `dashcoeff` or "alias" or
+            "formal" to run.
+        dashparam : dict, optional
+            Values for the same keys as `dashcoeff[dashlvl]['default']`
+            used to override any or all values initialized by `func`.
```

Comment:
Given your comment below, this is not true for dftd4.

Reply:
I've revised the docstring, I hope truthfully. There were several principles that wanted to be followed here ((a) psi4 dft-d interface should be consistent, (b) qcengine harness should follow upstream project in keyword handling, (c) qcengine dash harnesses (dftd3, gcp, mp2d, dftd4) should be consistent, (d) Molecule.run_dftdN (aka psi4 -d interface) should be consistent), and I just couldn't make them all so. The `Molecule.run_dftdN` seemed like the least important one to sacrifice.